porting/build_and_boot/standalone_kernel_build.rst

```diff
@@ -0,0 +1,464 @@
+
+      * "dtb address" is sometimes the same as "kernel tags load address". But if it's different, you should use this for dtb and tags for tags. This value does require some magic modifications. To get the proper value for this, run "python3 -c "print(hex(<your "dtb address" value here>))" (obviously removing the < and >). For example, 
+
+    python3 -c "print(hex(0x0000000047880000))"
+
+    will print 0x47880000. This is your required value.
+
+      * "page size" is required for the ramdisk to know what your flash chip uses for page sizes. Value after colon is what you need. For example, 2048.
+
+      * "os version" is the value which determines which Android version this boot.img has. Some bootloaders enforce this. For example, 10.0.
+
+      * "os patch level" is a similar story. You'll also want to grab this value, for example, 2021-11.
+
+      * "command line args" go straight into deviceinfo_kernel_cmdline. Make sure to keep "console=tty0" in there. No examples here :(
+
+      * If you're building vendor_boot, this command will also output a ``bootconfig`` file in your output directory. Move this to your source tree's root.
+
+      * Apart from these mentioned values, you'll want to skip everything else.
+
+
+Now according to said guide, fill in your offsets:
+
+
+* Put the value from "page size" into this config.
+
+``deviceinfo_flash_pagesize="2048"``
+
+
+* Base offset will always remain 0x00000000.
+
+``deviceinfo_flash_offset_base="0x00000000"``
+
+
+* Fill this in from "kernel load address".
+
+``deviceinfo_flash_offset_kernel="0x40080000"``
+
+
+* This gets its value from "ramdisk load address".
+
+``deviceinfo_flash_offset_ramdisk="0x51b00000"``
+
+
+* Although not mission critical, if you did get its value, fill it in from "second bootloader load address".
+
+``deviceinfo_flash_offset_second="0xc0ff0000"``
+
+
+* Tags should be filled in from "kernel tags load address".
+
+``deviceinfo_flash_offset_tags="0x47880000"``
+
+
+* DTB offset comes from "dtb load address" after fixing it using the guide.
+
+``deviceinfo_flash_offset_dtb="0x47880000"``
+
+
+* This comes from "os version". Not exactly required but some OEMs enforce it.
+
+``deviceinfo_bootimg_os_version="10.0"``
+
+
+* You'll want to put the "os patch level" value here.
+
+``deviceinfo_bootimg_os_patch_level="2021-11"``
+
+
+* If you're building a vendor_boot and you have moved the ``bootconfig`` file to your tree's root, add the following:
+
+``deviceinfo_vendor_bootconfig_path="bootconfig"``
+
+
+And just like that, you've filled in your deviceinfo properly and can now get on with the port!
+    
+
+Building, installing and running
+--------------------------------
+
+After you've completed your deviceinfo and filled in all needed stuff, its time you get to the main part, the build. For this just run:
+``./build.sh -b out``
```

Comment:
better do change it to `./build.sh -b workdir` or similar, as out is used for some other stuff too.

Reply:
Done